bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -0,0 +1,218 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.pushbullet.internal.action.PushbulletActions;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.pushbullet.internal.PushbulletConfiguration;
+import org.openhab.binding.pushbullet.internal.model.Push;
+import org.openhab.binding.pushbullet.internal.model.PushResponse;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+/**
+ * The {@link PushbulletHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+@NonNullByDefault
+public class PushbulletHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PushbulletHandler.class);
+
+    private final Gson gson = new GsonBuilder().create();
+
+    private static final Version VERSION = FrameworkUtil.getBundle(PushbulletHandler.class).getVersion();
+
+    private static final Pattern CHANNEL_PATTERN = Pattern.compile("^[a-zA-Z0-9_-]+$");
+
+    private @Nullable PushbulletConfiguration config;
+
+    public PushbulletHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("About to handle {} on {}", command, channelUID);
+
+        // Future improvement: If recipient is already set, send a push on a command channel change
+        // check reconnect channel of the unifi binding for that
+
+        logger.debug("The Pushbullet binding is a read-only binding and cannot handle command '{}'.", command);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(PushbulletConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
```

Comment:
This comment conflicts with line 98

Reply:
True, was a leftover from the template code. Removed.